src/crepe/Resource.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace crepe {
+
+class ResourceManager;
+class Asset;
+
+/**
+ * \brief Resource interface
+ *
+ * Resource is an interface class used to represent a (deserialized) game resource (e.g.
+ * textures, sounds). Resources are always created from \ref Asset "assets" by ResourceManager.
+ *
+ * The game programmer has the ability to use the ResourceManager to keep instances of concrete
+ * resources between scenes, preventing them from being reinstantiated during a scene
+ * transition.
+ */
+class Resource {
+public:
+	Resource(const Asset & src);
```

Comment:
ik ziet het nut niet van een resource constructor kan die niet default en de cpp verwijderen

Reply:
Het liefst zou ik de constructor een pure virtual function maken maar dat kan niet in C++ dus hij staat hier om aan te geven dat concrete resource klassen deze zelf moeten implementeren.